pandasai/smart_datalake/__init__.py

```diff
@@ -30,167 +28,87 @@
-from ..helpers.df_info import DataFrameType
 from ..helpers.path import find_project_root
-from ..helpers.viz_library_types.base import VisualizationLibrary
+import pandas as pd
 
 
 class SmartDatalake:
-    _dfs: List[DataFrameType]
-    _config: Union[Config, dict]
-    _llm: LLM
-    _cache: Cache = None
-    _logger: Logger
-    _last_prompt_id: uuid.UUID
-    _conversation_id: uuid.UUID
-    _code_manager: CodeManager
-    _memory: Memory
-    _skills: SkillsManager
-    _instance: str
-    _query_exec_tracker: QueryExecTracker
-
-    _last_code_generated: str = None
-    _last_result: str = None
-    _last_error: str = None
-
-    _viz_lib: str = None
-
     def __init__(
         self,
-        dfs: List[Union[DataFrameType, Any]],
+        dfs: List[Union[pd.DataFrame, Any]],
         config: Optional[Union[Config, dict]] = None,
         logger: Logger = None,
         memory: Memory = None,
         cache: Cache = None,
     ):
         """
         Args:
-            dfs (List[Union[DataFrameType, Any]]): List of dataframes to be used
+            dfs (List[Union[pd.DataFrame, Any]]): Pandas dataframe
             config (Union[Config, dict], optional): Config to be used. Defaults to None.
             logger (Logger, optional): Logger to be used. Defaults to None.
         """
+        self.last_prompt = None
+        self.last_prompt_id = None
+        self.last_result = None
+        self.last_code_generated = None
+        self.last_code_executed = None
 
-        self._load_config(config)
-
-        self.initialize()
-
-        if logger:
-            self.logger = logger
-        else:
-            self.logger = Logger(
-                save_logs=self._config.save_logs, verbose=self._config.verbose
-            )
+        config = self.load_config(config)
 
-        self._load_dfs(dfs)
-
-        self._memory = memory or Memory()
-        self._code_manager = CodeManager(
-            dfs=self._dfs,
-            config=self._config,
-            logger=self.logger,
+        self.logger = logger or Logger(
+            save_logs=config.save_logs, verbose=config.verbose
         )
+        dfs = self.load_dfs(dfs, config)
 
-        self._skills = SkillsManager()
-
-        if cache:
-            self._cache = cache
-        elif self._config.enable_cache:
-            self._cache = Cache()
-
-        context = Context(self._config, self.logger, self.engine)
+        self.conversation_id = uuid.uuid4()
+        self.instance = self.__class__.__name__
 
-        if self._config.response_parser:
-            self._response_parser = self._config.response_parser(context)
-        else:
-            self._response_parser = ResponseParser(context)
-
-        if self._config.data_viz_library:
-            self._viz_lib = self._config.data_viz_library.value
-
-        self._conversation_id = uuid.uuid4()
-
-        self._instance = self.__class__.__name__
+        query_exec_tracker = QueryExecTracker(
```

Comment:
@gventuri  We can move this initialization to PipelineContext

Reply:
@ArslanSaleem great catch, but that's part of another refactor that I'd like to keep separated (which involves logging a new event on the query logger for each step of the pipeline. By that time, the whole QueryTracker logic will be part of the pipelines